actors/builtin/miner/miner_actor.go

```diff
@@ -1366,25 +1366,35 @@ func (a Actor) ReportConsensusFault(rt Runtime, params *ReportConsensusFaultPara
 		rt.Abortf(exitcode.ErrIllegalArgument, "invalid fault epoch %v ahead of current %v", fault.Epoch, rt.CurrEpoch())
 	}
 
-	// Reward reporter with a share of the miner's current balance.
-	slasherReward := RewardForConsensusSlashReport(faultAge, rt.CurrentBalance())
-	_, code := rt.Send(reporter, builtin.MethodSend, nil, slasherReward)
-	builtin.RequireSuccess(rt, code, "failed to reward reporter")
-
+	// Penalize miner consensus fault fee
+	// Give a portion of this to the reporter as reward
 	var st State
-	rt.State().Readonly(&st)
-
-	// Notify power actor with lock-up total being removed.
-	_, code = rt.Send(
-		builtin.StoragePowerActorAddr,
-		builtin.MethodsPower.OnConsensusFault,
-		&st.LockedFunds,
-		abi.NewTokenAmount(0),
-	)
-	builtin.RequireSuccess(rt, code, "failed to notify power actor on consensus fault")
+	rewardStats := requestCurrentEpochBlockReward(rt)
+	faultPenalty := ConsensusFaultPenalty(rewardStats.ThisEpochReward)
+	slasherReward := RewardForConsensusSlashReport(faultAge, faultPenalty)
+	pledgeDelta := big.Zero()
+	rt.State().Transaction(&st, func() {
+		unlockedBalance := st.GetUnlockedBalance(rt.CurrentBalance())
+		penaltyFromVesting, penaltyFromBalance, err := st.PenalizeFundsInPriorityOrder(adt.AsStore(rt), rt.CurrEpoch(), faultPenalty, unlockedBalance)
+		builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to unlock unvested funds")
+		faultPenalty = big.Add(penaltyFromVesting, penaltyFromBalance)
```

Comment:
Either signpost this with a comment about how it's updating to be the amount actually payable, which might be less, or use a new variable name.